brainglobe_template_builder/preproc/transform_utils.py

```diff
@@ -71,3 +73,60 @@ def apply_transform(
     # Preserve original data range and type
     transformed = np.clip(transformed, data.min(), data.max())
     return transformed.astype(data.dtype)
+
+
+def downsample_anisotropic_image_stack(
+    stack: da.Array, xy_downsampling: int, z_downsampling: int
+) -> np.ndarray:
+    """
+
+    Lazily downsamples a dask array first along axis 1,2 and then along axis 0,
+    using a local mean of the pixels. The (smaller) array is returned
+    in memory (numpy) form at the end.
```

Comment:
Maybe explicitly state that the axis order is zyx (is zxy also valid?), though this is kinda implied from your description.